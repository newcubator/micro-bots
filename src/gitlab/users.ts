import axios from "axios";
import { GitlabUser } from "./gitlab";
import { GITLAB_TOKEN } from "./token";

export const getMembers = async (groupId: string): Promise<GitlabUser[]> => {
  const users = await axios.get(`https://gitlab.com/api/v4/groups/${groupId}/members/all`, {
    headers: {
      Authorization: `Bearer ${GITLAB_TOKEN}`,
    },
    params: {
      per_page: 100,
    },
  });
  return users.data;
};
