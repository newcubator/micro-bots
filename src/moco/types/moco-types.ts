interface genericIdString {
    id: string;
    name: string;
}

export interface MocoUserType {
    id: string;
    firstname: string;
    lastname: string;
    active: boolean;
    external: boolean;
    email: string;
    mobile_phone: string;
    work_phone: string;
    home_address: string;
    info: string;
    birthday: string;
    avatar_url: string;
    custom_properties: any;
    unit: any;
    created_at: string;
    updated_at: string;
}

export interface MocoSchedule {
    id: number;
    date: string;
    comment: string;
    am: boolean;
    pm: boolean;
    symbol: number;
    assignment: {
        id: number;
        name: string;
        customer_name: string;
        color: string;
        type: string;
    };
    user: MocoUserType;
    created_at: string;
    updated_at: string;
}

export interface MocoEmployment {
    id: number;
    weekly_target_hours: number;
    pattern: {
        am: number[];
        pm: number[];
    };
    from: string;
    to: string;
    user: MocoUserType;
    created_at: string;
    updated_at: string;
}

export interface MocoActivity {
    id: number;
    date: string;
    hours: number;
    description: string;
    billed: boolean;
    billable: boolean;
    tag: string;
    remote_service: string;
    remote_id: string;
    remote_url: string;
    project: {
        id: number;
        name: string;
        billable: boolean;
    };
    task: {
        id: number;
        name: string;
        billable: boolean;
    };
    customer: genericIdString;
    user: MocoUserType;
    timer_started_at: string;
    created_at: string;
    updated_at: string;
    hourly_rate: number;
}

export interface MocoProject {
    id: number;
    identifier: string;
    name: string;
    active: boolean;
    billable: boolean;
    fixed_price: boolean;
    finish_date: Date;
    currency: string;
    billing_variant: string;
    billing_address: string;
    billing_email_to: null;
    billing_email_cc: null;
    billing_notes: null;
    setting_include_time_report: boolean;
    budget: number;
    budget_expenses: number;
    hourly_rate: number;
    info: string;
    tags: any[];
    labels: any[];
    custom_properties: {
        Bestellnummer: string;
    };
    leader: Leader;
    customer: genericIdString;
    tasks: Task[];
    contracts: MocoContract[];
    deal: genericIdString;
    created_at: Date;
    updated_at: Date;
}

export interface MocoContract {
    id: number;
    user_id: number;
    firstname: string;
    lastname: string;
    billable: boolean;
    active: boolean;
    budget: null;
    hourly_rate: number;
}

export interface Leader {
    id: number;
    firstname: string;
    lastname: string;
}

export interface Task {
    id: number;
    name: string;
    billable: boolean;
    active: boolean;
    budget: null;
    hourly_rate: number;
}

export interface MocoDeal {
    id: number;
    name: string;
    status: string;
    reminder_date: Date;
    closed_on: Date;
    money: number;
    currency: string;
    info: string;
    tags: any[];
    custom_properties: {};
    user: {
        id: string;
        firstname: string;
        lastname: string;
    };
    person: genericIdString;
    customer: genericIdString;
    company: {
        id: string;
        type: string;
        name: string;
    };
    category: {
        id: number;
        name: string;
        probability: number;
    };
    created_at: Date;
    updated_at: Date;
}

export interface MocoContact {
    id: number,
    gender: 'F' | 'H',
    firstname: string,
    lastname: string,
    title: string,
    job_position: string,
    mobile_phone: string,
    work_fax: string,
    work_phone: string,
    work_email: string,
    work_address: string,
    home_email: string,
    home_address: string,
    birthday: string,
    avatar_url: string,
    tags: [
        string
    ],
    custom_properties: {},
    company: {
        id: string;
        type: string;
        name: string;
    };
    user: {
        id: string,
        firstname: string,
        lastname: string
    },
    created_at: string,
    updated_at: string,
    info: string
}
