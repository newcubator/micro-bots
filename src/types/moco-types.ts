import {AxiosResponse} from "axios";

export interface MocoUserType {
    id: string;
    firstname: string;
    lastname: string;
    active: boolean;
    external: boolean;
    email: string;
    mobile_phone: string;
    work_phone: string;
    home_address: string;
    info: string;
    birthday: string;
    avatar_url: string;
    custom_properties: any;
    unit: any;
    created_at: string;
    updated_at: string;
}


export interface MocoSchedulesResponse extends AxiosResponse {
    data: MocoSchedule[];
}

export interface MocoSchedule {
    id: number;
    date: string;
    comment: string;
    am: boolean;
    pm: boolean;
    symbol: number;
    assignment: {
        id: number;
        name: string;
        customer_name: string;
        color: string;
        type: string;
    },
    user: MocoUserType;
    created_at: string;
    updated_at: string;
}

export interface MocoEmploymentsResponse extends AxiosResponse {
    data: MocoEmployment[];
}

export interface MocoEmployment {
    id: number,
    weekly_target_hours: number,
    pattern: { am: number[], pm: number[] },
    from: string,
    to: string,
    user: MocoUserType,
    created_at: string,
    updated_at: string
}

export interface MocoActivityResponse extends AxiosResponse {
    data: MocoActivity[];
}

export interface MocoActivity {
    id: number,
    date: string,
    hours: number,
    description: string,
    billed: boolean,
    billable: boolean,
    tag: string,
    remote_service: string,
    remote_id: string,
    remote_url: string,
    project: {
        id: number,
        name: string,
        billable: boolean
    },
    task: { id: number, name: string, billable: boolean },
    customer: { id: number, name: string },
    user: MocoUserType,
    timer_started_at: string,
    created_at: string,
    updated_at: string,
    hourly_rate: number

}

