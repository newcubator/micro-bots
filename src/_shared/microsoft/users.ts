import axios from "axios";
import { getAzureAccessToken } from "./token";
import { MicrosoftUsersType } from "./types/microsoft-types";

export async function getUsers(): Promise<MicrosoftUsersType> {
  const accessToken = await getAzureAccessToken();

  return axios
    .get(`https://graph.microsoft.com/v1.0/users/`, {
      headers: {
        Authorization: `Bearer ${accessToken}`,
      },
    })
    .then((response) => {
      return response.data;
    })
    .catch((error) => {
      console.log(error);
    });
}
