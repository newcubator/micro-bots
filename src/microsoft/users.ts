import axios from "axios";
import { getAzureAccessToken } from "./token";
import { MicrosoftUsersType } from "./types/microsoft-types";

export async function getUsers(): Promise<MicrosoftUsersType> {
  const azureAccessToken = await getAzureAccessToken();

  return axios
    .get(`https://graph.microsoft.com/v1.0/users/`, {
      headers: { Authorization: `Bearer ${azureAccessToken.access_token}` },
    })
    .then((response) => {
      return response.data;
    })
    .catch((error) => {
      console.log(error);
    });
}
