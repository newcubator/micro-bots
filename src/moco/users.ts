import axios, { AxiosResponse } from 'axios';
import { SlackCommandType } from '../slack/types/slack-types';
import { MOCO_TOKEN } from './token';
import { MocoUserType } from './types/moco-types';

/*
 * @See https://github.com/hundertzehn/mocoapp-api-docs/blob/master/sections/users.md
 */

export function getUsers(): Promise<Array<MocoUserType>> {
    return axios.get('https://newcubator.mocoapp.com/api/v1/users', {
        headers: {
            'Authorization': 'Token token=' + MOCO_TOKEN
        },
    }).then((response: AxiosResponse<Array<MocoUserType>>) => response.data);
}

export function findUserBySlackCommand(command: SlackCommandType): (users: Array<MocoUserType>) => MocoUserType {
    return (users: Array<MocoUserType>): MocoUserType => {
        let user: MocoUserType;
        user = findUserBySlackId(users, command.user_id);
        if (!user) {
            user = findUserByMailPrefix(users, command.user_name);
        }
        return user;
    };
}

function findUserBySlackId(users: Array<MocoUserType>, slackId: string): MocoUserType {
    return users.find(user => user.custom_properties.SlackId == slackId);
}

function findUserByMailPrefix(users: Array<MocoUserType>, prefix: string): MocoUserType {
    return users.find(user => user.email.startsWith(prefix));
}
