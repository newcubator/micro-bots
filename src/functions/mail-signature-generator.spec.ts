import { MocoUserType } from "../moco/types/moco-types";
import { findUserBySlackCommand, getUsers } from "../moco/users";
import { handler } from "./mail-signature-generator";
jest.mock("../moco/users");

const mockGetUsers = getUsers as unknown as jest.Mock;
const mockFindUserBySlackCommand = findUserBySlackCommand as unknown as jest.Mock;

describe("MailSignatureGenerator", () => {
  it("should generate a mail signature with hannover address", async () => {
    mockGetUsers.mockResolvedValue([]);
    mockFindUserBySlackCommand.mockReturnValueOnce(() => ({
      firstname: "Jörg",
      lastname: "Görisch",
      email: "joerg.goerisch@newcubator.com",
      custom_properties: {
        ["Job Bezeichnung"]: "Testsubject",
        Standort: "Hannover",
      },
    }));

    const result = await handler({
      queryStringParameters: { user_id: "U0113HJ8N2Z", user_name: "jan.sauer" },
    } as any);

    expect(result.statusCode).toBe(200);
    expect(result.headers).toEqual({
      "Content-Type": "text/html;charset=utf-8",
    });
    expect(result.body).toMatchSnapshot();
  });

  it("should generate a mail signature with dortmund address", async () => {
    mockGetUsers.mockResolvedValue([]);
    mockFindUserBySlackCommand.mockReturnValueOnce(() => ({
      firstname: "Jörg",
      lastname: "Görisch",
      email: "joerg.goerisch@newcubator.com",
      custom_properties: {
        ["Job Bezeichnung"]: "Testsubject",
        Standort: "Dortmund",
      },
    }));

    const result = await handler({
      queryStringParameters: { user_id: "U0113HJ8N2Z", user_name: "jan.sauer" },
    } as any);

    expect(result.statusCode).toBe(200);
    expect(result.headers).toEqual({
      "Content-Type": "text/html;charset=utf-8",
    });
    expect(result.body).toMatchSnapshot();
  });

  it("should return a useful message if the user was not found", async () => {
    mockGetUsers.mockResolvedValue([]);
    mockFindUserBySlackCommand.mockReturnValueOnce(() => null);

    const result = await handler({
      queryStringParameters: { user_id: "U0113HJ8N2Z", user_name: "jan.sauer" },
    } as any);

    expect(result.statusCode).toBe(200);
    expect(result.body).toEqual("Ich konnte dich leider keinem Moco User zuordnen.");
  });
});
