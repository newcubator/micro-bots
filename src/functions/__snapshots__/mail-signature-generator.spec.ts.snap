// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailSignatureGenerator should generate a mail signature with dortmund address 1`] = `
"<body>
<div id=\\"signature\\">
<table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
  <tbody>
    <tr>
      <td>
        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
          <tbody>
            <tr>
              <td style=\\"vertical-align: top;\\">
                <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                  <tbody>
                    <tr>
                      <td style=\\"text-align: center;\\">
                        <img src=\\"https://newcubator.com/images/mailsignature/joerg-goerisch.png\\" role=\\"presentation\\" width=\\"130\\" style=\\"max-width: 128px; display: block;\\">
                      </td>
                    </tr>
                    <tr>
                      <td height=\\"30\\"></td>
                    </tr>
                      <tr>
                        <td style=\\"text-align: center;\\">
                          <img src=\\"https://newcubator.com/images/mailsignature/nc-logo.png\\" role=\\"presentation\\" width=\\"130\\" style=\\"max-width: 130px; display: block;\\">
                        </td>
                      </tr>
                    <tr>
                      <td height=\\"30\\"></td>
                    </tr>
                    <tr>
                      <td style=\\"text-align: center;\\">
                        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial; display: inline-block;\\">
                          <tbody>
                            <tr style=\\"text-align: center;\\">
                              <td>
                                <a href=\\"https://twitter.com/newcubator\\" color=\\"#50505E\\" style=\\"display: inline-block; padding: 0px; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/twitter-icon-2x.png\\" alt=\\"twitter\\" color=\\"#50505E\\" height=\\"24\\" style=\\"background-color: rgba(255, 255, 255, 0); max-width: 135px; display: block;\\">
                                </a>
                              </td>
                              <td width=\\"5\\">
                                <div></div>
                              </td>
                              <td>
                                <a href=\\"https://www.linkedin.com/company/newcubator/\\" color=\\"#50505E\\" style=\\"display: inline-block; padding: 0px; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/linkedin-icon-2x.png\\" alt=\\"linkedin\\" color=\\"#50505E\\" height=\\"24\\" style=\\"background-color: rgba(255, 255, 255, 0); max-width: 135px; display: block;\\">
                                </a>
                              </td>
                              <td width=\\"5\\">
                                <div></div>
                              </td>
                              <td>
                                <a href=\\"https://www.instagram.com/newcubator/?hl=de\\" color=\\"#50505E\\" style=\\"display: inline-block; padding: 0px; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/instagram-icon-2x.png\\" alt=\\"instagram\\" color=\\"#50505E\\" height=\\"24\\" style=\\"background-color: rgba(255, 255, 255, 0); max-width: 135px; display: block;\\">
                                </a>
                              </td>
                              <td width=\\"5\\">
                                <div></div>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </td>
              <td width=\\"46\\">
                <div></div>
              </td>
              <td style=\\"padding: 0px; vertical-align: middle;\\">
                <h3 color=\\"#50505e\\" style=\\"margin: 0px; font-size: 18px; color: rgb(80, 80, 94);\\">
                  <span>J&ouml;rg</span> <span>G&ouml;risch</span>
                </h3>
                <p color=\\"#50505e\\" font-size=\\"medium\\" style=\\"margin: 0px; font-weight: 500; color: rgb(80, 80, 94); font-size: 14px; line-height: 22px;\\">
                  <span>Testsubject</span><span>&nbsp;|&nbsp;</span><span>newcubator GmbH</span>
                </p>
                <p color=\\"#50505e\\" font-size=\\"medium\\" style=\\"color: rgb(80, 80, 94); margin: 0px; font-size: 14px; line-height: 22px;\\">
                  <span>Gesch&auml;ftsf&uuml;hrer: J&ouml;rg Herbst</span>
                </p>
                <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial; width: 100%;\\">
                  <tbody>
                    <tr>
                      <td height=\\"30\\"></td>
                    </tr>
                    <tr>
                      <td color=\\"#405780\\" direction=\\"horizontal\\" height=\\"1\\" style=\\"width: 100%; border-bottom: 1px solid rgb(64, 87, 128); border-left: none; display: block;\\"></td>
                    </tr>
                    <tr>
                      <td height=\\"30\\"></td>
                    </tr>
                  </tbody>
                </table>
                <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                  <tbody>
                    <tr height=\\"25\\" style=\\"vertical-align: middle;\\">
                      <td width=\\"30\\" style=\\"vertical-align: middle;\\">
                        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                          <tbody>
                            <tr>
                              <td style=\\"vertical-align: bottom;\\">
                                <span color=\\"#405780\\" width=\\"11\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/phone-icon-2x.png\\" color=\\"#405780\\" width=\\"13\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                </span>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                      <td style=\\"padding: 0px; color: rgb(80, 80, 94);\\">
                        <a href=\\"tel:+49 231 58687380\\" color=\\"#50505e\\" style=\\"text-decoration: none; color: rgb(80, 80, 94); font-size: 12px;\\">
                          <span>+49 231 58687380</span>
                        </a>
                      </td>
                    </tr>
                    <tr height=\\"25\\" style=\\"vertical-align: middle;\\">
                      <td width=\\"30\\" style=\\"vertical-align: middle;\\">
                        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                          <tbody>
                            <tr>
                              <td style=\\"vertical-align: bottom;\\">
                                <span color=\\"#405780\\" width=\\"11\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/email-icon-2x.png\\" color=\\"#405780\\" width=\\"13\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                </span>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                      <td style=\\"padding: 0px;\\">
                        <a href=\\"mailto:joerg.goerisch@newcubator.com\\" color=\\"#50505e\\" style=\\"text-decoration: none; color: rgb(80, 80, 94); font-size: 12px;\\">
                          <span>joerg.goerisch@newcubator.com</span>
                        </a>
                      </td>
                    </tr>
                    <tr height=\\"25\\" style=\\"vertical-align: middle;\\">
                      <td width=\\"30\\" style=\\"vertical-align: middle;\\">
                        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                          <tbody>
                            <tr>
                              <td style=\\"vertical-align: bottom;\\">
                                <span color=\\"#405780\\" width=\\"11\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/link-icon-2x.png\\" color=\\"#405780\\" width=\\"13\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                </span>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                      <td style=\\"padding: 0px;\\">
                        <a href=\\"//newcubator.com\\" color=\\"#50505e\\" style=\\"text-decoration: none; color: rgb(80, 80, 94); font-size: 12px;\\">
                          <span>newcubator.com</span>
                        </a>
                      </td>
                    </tr>
                    <tr height=\\"25\\" style=\\"vertical-align: middle;\\">
                      <td width=\\"30\\" style=\\"vertical-align: middle;\\">
                        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                          <tbody>
                            <tr>
                              <td style=\\"vertical-align: bottom;\\">
                                <span color=\\"#405780\\" width=\\"11\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/address-icon-2x.png\\" color=\\"#405780\\" width=\\"13\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                </span>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                      <td style=\\"padding: 0px;\\">
                        <span color=\\"#50505e\\" style=\\"font-size: 12px; color: rgb(80, 80, 94);\\">
                          <span>Westenhellweg 85-89, 44137 Dortmund, Amtsgericht Hannover HRB 221930</span>
                        </span>
                      </td>
                    </tr>
                  </tbody>
                </table>
                <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                  <tbody>
                    <tr>
                      <td height=\\"30\\"></td>
                    </tr>
                  </tbody>
                </table>
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
    <tr>
      <td>
        <a href=\\"https://newcubator.com/email-marketing-banner\\" style=\\"display: inline-block; padding: 0; width: 593px; background-color: rgba(255, 255, 255, 0);\\">
          <img src=\\"https://newcubator.com/images/email-marketing-banner/email-footer-image.jpg\\" alt=\\"newcubator banner\\" style=\\"display: block; width: 593px;\\">
        </a>
      </td>
    </tr>
  </tbody>
</table>
</div>
    <button onclick=\\"copy()\\">Copy to Clipboard</button>
</body>
<script type=\\"text/javascript\\">
function replaceUmlautsWithHtml(str) {
  return str
    .replace(/ä/g, \\"&auml;\\")
    .replace(/ö/g, \\"&ouml;\\")
    .replace(/ü/g, \\"&uuml;\\")
    .replace(/ß/g, \\"&szlig;\\")
    .replace(/Ä/g, \\"&Auml;\\")
    .replace(/Ö/g, \\"&Ouml;\\")
    .replace(/Ü/g, \\"&Uuml;\\");
}

function copy() {
    navigator.clipboard.writeText(replaceUmlautsWithHtml(document.getElementById('signature').innerHTML));
}
</script>"
`;

exports[`MailSignatureGenerator should generate a mail signature with hannover address 1`] = `
"<body>
<div id=\\"signature\\">
<table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
  <tbody>
    <tr>
      <td>
        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
          <tbody>
            <tr>
              <td style=\\"vertical-align: top;\\">
                <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                  <tbody>
                    <tr>
                      <td style=\\"text-align: center;\\">
                        <img src=\\"https://newcubator.com/images/mailsignature/joerg-goerisch.png\\" role=\\"presentation\\" width=\\"130\\" style=\\"max-width: 128px; display: block;\\">
                      </td>
                    </tr>
                    <tr>
                      <td height=\\"30\\"></td>
                    </tr>
                      <tr>
                        <td style=\\"text-align: center;\\">
                          <img src=\\"https://newcubator.com/images/mailsignature/nc-logo.png\\" role=\\"presentation\\" width=\\"130\\" style=\\"max-width: 130px; display: block;\\">
                        </td>
                      </tr>
                    <tr>
                      <td height=\\"30\\"></td>
                    </tr>
                    <tr>
                      <td style=\\"text-align: center;\\">
                        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial; display: inline-block;\\">
                          <tbody>
                            <tr style=\\"text-align: center;\\">
                              <td>
                                <a href=\\"https://twitter.com/newcubator\\" color=\\"#50505E\\" style=\\"display: inline-block; padding: 0px; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/twitter-icon-2x.png\\" alt=\\"twitter\\" color=\\"#50505E\\" height=\\"24\\" style=\\"background-color: rgba(255, 255, 255, 0); max-width: 135px; display: block;\\">
                                </a>
                              </td>
                              <td width=\\"5\\">
                                <div></div>
                              </td>
                              <td>
                                <a href=\\"https://www.linkedin.com/company/newcubator/\\" color=\\"#50505E\\" style=\\"display: inline-block; padding: 0px; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/linkedin-icon-2x.png\\" alt=\\"linkedin\\" color=\\"#50505E\\" height=\\"24\\" style=\\"background-color: rgba(255, 255, 255, 0); max-width: 135px; display: block;\\">
                                </a>
                              </td>
                              <td width=\\"5\\">
                                <div></div>
                              </td>
                              <td>
                                <a href=\\"https://www.instagram.com/newcubator/?hl=de\\" color=\\"#50505E\\" style=\\"display: inline-block; padding: 0px; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/instagram-icon-2x.png\\" alt=\\"instagram\\" color=\\"#50505E\\" height=\\"24\\" style=\\"background-color: rgba(255, 255, 255, 0); max-width: 135px; display: block;\\">
                                </a>
                              </td>
                              <td width=\\"5\\">
                                <div></div>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </td>
              <td width=\\"46\\">
                <div></div>
              </td>
              <td style=\\"padding: 0px; vertical-align: middle;\\">
                <h3 color=\\"#50505e\\" style=\\"margin: 0px; font-size: 18px; color: rgb(80, 80, 94);\\">
                  <span>J&ouml;rg</span> <span>G&ouml;risch</span>
                </h3>
                <p color=\\"#50505e\\" font-size=\\"medium\\" style=\\"margin: 0px; font-weight: 500; color: rgb(80, 80, 94); font-size: 14px; line-height: 22px;\\">
                  <span>Testsubject</span><span>&nbsp;|&nbsp;</span><span>newcubator GmbH</span>
                </p>
                <p color=\\"#50505e\\" font-size=\\"medium\\" style=\\"color: rgb(80, 80, 94); margin: 0px; font-size: 14px; line-height: 22px;\\">
                  <span>Gesch&auml;ftsf&uuml;hrer: J&ouml;rg Herbst</span>
                </p>
                <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial; width: 100%;\\">
                  <tbody>
                    <tr>
                      <td height=\\"30\\"></td>
                    </tr>
                    <tr>
                      <td color=\\"#405780\\" direction=\\"horizontal\\" height=\\"1\\" style=\\"width: 100%; border-bottom: 1px solid rgb(64, 87, 128); border-left: none; display: block;\\"></td>
                    </tr>
                    <tr>
                      <td height=\\"30\\"></td>
                    </tr>
                  </tbody>
                </table>
                <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                  <tbody>
                    <tr height=\\"25\\" style=\\"vertical-align: middle;\\">
                      <td width=\\"30\\" style=\\"vertical-align: middle;\\">
                        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                          <tbody>
                            <tr>
                              <td style=\\"vertical-align: bottom;\\">
                                <span color=\\"#405780\\" width=\\"11\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/phone-icon-2x.png\\" color=\\"#405780\\" width=\\"13\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                </span>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                      <td style=\\"padding: 0px; color: rgb(80, 80, 94);\\">
                        <a href=\\"tel:+49 511 95731300\\" color=\\"#50505e\\" style=\\"text-decoration: none; color: rgb(80, 80, 94); font-size: 12px;\\">
                          <span>+49 511 95731300</span>
                        </a>
                      </td>
                    </tr>
                    <tr height=\\"25\\" style=\\"vertical-align: middle;\\">
                      <td width=\\"30\\" style=\\"vertical-align: middle;\\">
                        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                          <tbody>
                            <tr>
                              <td style=\\"vertical-align: bottom;\\">
                                <span color=\\"#405780\\" width=\\"11\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/email-icon-2x.png\\" color=\\"#405780\\" width=\\"13\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                </span>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                      <td style=\\"padding: 0px;\\">
                        <a href=\\"mailto:joerg.goerisch@newcubator.com\\" color=\\"#50505e\\" style=\\"text-decoration: none; color: rgb(80, 80, 94); font-size: 12px;\\">
                          <span>joerg.goerisch@newcubator.com</span>
                        </a>
                      </td>
                    </tr>
                    <tr height=\\"25\\" style=\\"vertical-align: middle;\\">
                      <td width=\\"30\\" style=\\"vertical-align: middle;\\">
                        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                          <tbody>
                            <tr>
                              <td style=\\"vertical-align: bottom;\\">
                                <span color=\\"#405780\\" width=\\"11\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/link-icon-2x.png\\" color=\\"#405780\\" width=\\"13\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                </span>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                      <td style=\\"padding: 0px;\\">
                        <a href=\\"//newcubator.com\\" color=\\"#50505e\\" style=\\"text-decoration: none; color: rgb(80, 80, 94); font-size: 12px;\\">
                          <span>newcubator.com</span>
                        </a>
                      </td>
                    </tr>
                    <tr height=\\"25\\" style=\\"vertical-align: middle;\\">
                      <td width=\\"30\\" style=\\"vertical-align: middle;\\">
                        <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                          <tbody>
                            <tr>
                              <td style=\\"vertical-align: bottom;\\">
                                <span color=\\"#405780\\" width=\\"11\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                  <img src=\\"https://newcubator.com/images/mailsignature/address-icon-2x.png\\" color=\\"#405780\\" width=\\"13\\" style=\\"display: block; background-color: rgba(255, 255, 255, 0);\\">
                                </span>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                      <td style=\\"padding: 0px;\\">
                        <span color=\\"#50505e\\" style=\\"font-size: 12px; color: rgb(80, 80, 94);\\">
                          <span>B&ouml;dekerstra&szlig;e 22, 30161 Hannover, Amtsgericht Hannover HRB 221930</span>
                        </span>
                      </td>
                    </tr>
                  </tbody>
                </table>
                <table cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"vertical-align: -webkit-baseline-middle; font-size: medium; font-family: Arial;\\">
                  <tbody>
                    <tr>
                      <td height=\\"30\\"></td>
                    </tr>
                  </tbody>
                </table>
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
    <tr>
      <td>
        <a href=\\"https://newcubator.com/email-marketing-banner\\" style=\\"display: inline-block; padding: 0; width: 593px; background-color: rgba(255, 255, 255, 0);\\">
          <img src=\\"https://newcubator.com/images/email-marketing-banner/email-footer-image.jpg\\" alt=\\"newcubator banner\\" style=\\"display: block; width: 593px;\\">
        </a>
      </td>
    </tr>
  </tbody>
</table>
</div>
    <button onclick=\\"copy()\\">Copy to Clipboard</button>
</body>
<script type=\\"text/javascript\\">
function replaceUmlautsWithHtml(str) {
  return str
    .replace(/ä/g, \\"&auml;\\")
    .replace(/ö/g, \\"&ouml;\\")
    .replace(/ü/g, \\"&uuml;\\")
    .replace(/ß/g, \\"&szlig;\\")
    .replace(/Ä/g, \\"&Auml;\\")
    .replace(/Ö/g, \\"&Ouml;\\")
    .replace(/Ü/g, \\"&Uuml;\\");
}

function copy() {
    navigator.clipboard.writeText(replaceUmlautsWithHtml(document.getElementById('signature').innerHTML));
}
</script>"
`;
